windows/devices/uwb/include/windows/devices/uwb/UwbDeviceConnector.hxx

```diff
@@ -48,14 +50,31 @@ public:
      * @return false If listening for notifications could not be started.
      */
     bool
-    NotificationListenerStart(std::function<void(::uwb::protocol::fira::UwbNotificationData)> onNotification);
+    NotificationListenerStart();
 
     /**
      * @brief Stop listening for notifications.
      */
     void
     NotificationListenerStop();
 
+    /**
+     * @brief Registers the callbacks for a particular session
+     *
+     * @param sessionId
+     * @param callbacks
+     */
+    void
+    RegisterSessionEventCallbacks(uint32_t sessionId, std::weak_ptr<::uwb::UwbSessionEventCallbacks> callbacks);
+
+    /**
+     * @brief Sets the callbacks for the UwbDevice that owns this UwbDeviceConnector
+     *
+     * @param callbacks
+     */
+    void
+    RegisterDeviceEventCallbacks(::uwb::UwbDeviceEventCallbacks callbacks);

```

Comment:
I really like this interface, it's simple and accomplishes exactly what we need.